third_party/amd/lib/TritonAMDGPUTransforms/StreamPipelineV2.cpp

```diff
@@ -207,9 +207,21 @@ getSharedEncIfAllUsersAreDotEnc(Value val) {
       auto CTALayout = ttg::getCTALayout(srcTy.getEncoding());
       auto order = ttg::getOrder(srcTy.getEncoding());
       unsigned bitWidth = srcTy.getElementType().getIntOrFloatBitWidth();
+      SmallVector<unsigned> sharedOrder;
+      int rank = order.size();
+      // TODO rework this when shared -> dotOp conversions support arbitrary
+      // shared memory ordering
+      if (rank == 3) {
+        // add all elements except the element that is zero
```

Comment:
The current comment just describes what the impl is doing in English; it's not very useful. What about something like "Move the batch dimension (dim #0) to be the last so that it will be the slowest varying dimension."